src/main/java/org/mitre/synthea/export/FhirStu3.java

```diff
@@ -810,6 +820,313 @@ private static BundleEntryComponent encounterClaim(BundleEntryComponent personEn
     return newEntry(bundle, claimResource);
   }
 
+
+  private static BundleEntryComponent explanationOfBenefit(BundleEntryComponent personEntry,
+                                           Bundle bundle, BundleEntryComponent encounterEntry,
+                                           Person person, org.hl7.fhir.dstu3.model.Claim claim,
+                                                           Encounter encounter) {
+
+    ExplanationOfBenefit eob = new ExplanationOfBenefit();
+    FhirContext fc3 = FhirContext.forDstu3();
+    org.hl7.fhir.dstu3.model.Encounter encounterResource =
+        (org.hl7.fhir.dstu3.model.Encounter) encounterEntry.getResource();
+
+
+
+    // according to CMS guidelines claims have 12 months to be
+    // billed, so we set the billable period to 1 year after
+    // services have ended (the encounter ends).
+    Calendar cal = Calendar.getInstance();
+    cal.setTime(encounterResource.getPeriod().getEnd());
+    cal.add(Calendar.YEAR,1);
+
+    Period billablePeriod = new Period()
+        .setStart(encounterResource
+            .getPeriod()
+            .getEnd())
+        .setEnd(cal.getTime());
+    eob.setBillablePeriod(billablePeriod);
+
+    // cost is hardcoded to be USD in claim so this should be fine as well
+    Money totalCost = new Money();
+    totalCost.setSystem("urn:iso:std:iso:4217");
+    totalCost.setCode("USD");
+    totalCost.setValue(encounter.claim.total());
+    eob.setTotalCost(totalCost);
+
+    // Set References
+    eob.setPatient(claim.getPatient());
+    eob.setOrganization(claim.getOrganization());
+    eob.setProvider(claim.getProvider());
+    eob.setReferral(new Reference("#1"));
+
+
+    eob.addIdentifier()
+        .setSystem("https://bluebutton.cms.gov/resources/variables/clm_id")
+        .setValue(UUID.randomUUID().toString());
```

Comment:
I believe this is supposed to be the claim id (e.g. `claim.getId()`) and not a random UUID... that being said, this is an abomination, since we should just do something like `eob.setClaim(claim.getId());` rather than use this CMS extension. That being said, you should do both.